cpp/src/arrow/compute/api_vector.h

```diff
@@ -174,6 +174,25 @@ class ARROW_EXPORT SelectKOptions : public FunctionOptions {
   std::vector<SortKey> sort_keys;
 };
 
+/// \brief Rank options
+class ARROW_EXPORT RankOptions : public FunctionOptions {
+ public:
+  enum Tiebreaker { Min, Max, First, Dense };
+
+  explicit RankOptions(SortOrder order = SortOrder::Ascending,
+                       NullPlacement null_placement = NullPlacement::AtEnd,
+                       Tiebreaker = RankOptions::First);
+  static constexpr char const kTypeName[] = "RankOptions";
+  static RankOptions Defaults() { return RankOptions(); }
+
+  /// Sorting order
+  SortOrder order;
```

Comment:
Do we want to be forward-looking? If at some point we want to implement multi-column ranking just like we have multi-column sorting, we may want this to be `std::vector<SortKey> sort_keys` instead.

We can still keep the current constructor which would initialize a single sort key with an empty field name, since that's sufficient for arrays.

Reply:
@lidavidm @kou What is your opinion on this?

Reply:
I like `std::vector<SortKey> sort_keys` here. If we need a shortcut for `Array`, we can add `ArrayRankOptions` like `ArraySortOptions`.

Reply:
Went ahead and did this. Right now with `ArrayRankOptions` I think we'd also have to register something like `array_rank` to actually be able to use it. Not sure if we want to implement that, keep things as is for consistency with `ArraySortOptions` or just remove `ArrayRankOptions`